snap/quota/quota.go

```diff
@@ -257,18 +259,39 @@ func ResolveCrossReferences(grps map[string]*Group) error {
 
 // tree recursively returns all of the sub-groups of the group and the group
 // itself.
-func (grp *Group) tree() []*Group {
-	// TODO: should we be paranoid about circular references and
-	// recursion here?
-	treeList := grp.subGroups
+func (grp *Group) tree(acc []*Group) ([]*Group, error) {
+	// make sure that none of the sub-groups seen already are present in the
+	// sub-groups of this group
 	for _, sub := range grp.subGroups {
-		treeList = append(treeList, sub.tree()...)
+		// check if we have already seen this sub-group
+		for _, alreadySeen := range acc {
```

Comment:
I think we should use a helper map to track already seen groups

Reply:
that's more typical yes

Reply:
done